aea/helpers/dialogue/base.py

```diff
@@ -543,13 +496,24 @@ def is_valid_next_message(self, message: Message) -> bool:
         :param message: the message to be validated
         :return: True if yes, False otherwise.
         """
-        return (
-            self._basic_rules(message)
-            and self._additional_rules(message)
-            and self.is_valid(message)
-        )
+        result_basic_validation, msg_basic_validation = self._basic_validation(message)
+        if not result_basic_validation:
+            return False, msg_basic_validation
+        else:
+            (
+                result_additional_validation,
+                msg_additional_validation,
+            ) = self._additional_validation(message)
+            if not result_additional_validation:
+                return False, msg_additional_validation
+            else:
+                result_is_valid, msg_is_valid = self.is_valid(message)
+                if not result_is_valid:
+                    return False, msg_is_valid
+                else:
+                    return True, "Message is valid with respect to this dialogue."
 
-    def _basic_rules(self, message: Message) -> bool:
+    def _basic_validation(self, message: Message) -> Tuple[bool, str]:
```

Comment:
Compare left and right; the code has blown up and become very hard to read. Firstly why? secondly, it needs to be done better.

Reply:
Not a fair comparison as previously you didn't know what's wrong with the message in case of an error, now there is detailed information as to which part failed.

Made some improvements (removed elif, elif, ...)